components/org.wso2.carbon.identity.user.endpoint/src/test/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateCodeApiServiceImplTest.java

```diff
@@ -43,14 +47,41 @@
     @Mock
     private UserSelfRegistrationManager userSelfRegistrationManager;
 
+    @Mock
+    private Throwable throwable;
+
     @InjectMocks
     private ValidateCodeApiServiceImpl validateCodeApiServiceImpl;
 
     @Test
-    public void testValidateCodePost() {
+    public void testValidateCodePost() throws Exception{
+
+        mockClasses();
+        assertEquals(validateCodeApiServiceImpl.validateCodePost(createCodeValidationRequestDTO()).getStatus(), 202);
+        assertEquals(validateCodeApiServiceImpl.validateCodePost(null).getStatus(), 202);
+    }
+    private void mockClasses() throws Exception {
+
 
         mockStatic(Utils.class);
         when(Utils.getUserSelfRegistrationManager()).thenReturn(userSelfRegistrationManager);
+    }
+
+    @Test
+    public void testIdentityRecoveryExceptioninResendCodePost() throws Exception {
+
+        mockClasses();
+        doThrow(new IdentityRecoveryException("Recovery Exception")).when(userSelfRegistrationManager).
+                confirmUserSelfRegistration(anyString());
```

Comment:
Should you not have expectedExceptions definition when we do doThrow?

@Test(expectedExceptions = IllegalStateException.class)

Reply:
Actually from ValidateCodeApiServiceImpl.java we have handled this exception by using a try catch block. So we this exception is not throwing to this level. So we don't need to expect the exception here